puppet/zulip/files/postgresql/process_fts_updates

```diff
@@ -80,51 +83,50 @@ def am_master(cursor: psycopg2.extensions.cursor) -> bool:
     return not cursor.fetchall()[0][0]
 
 
-pg_args = {}
+def get_config(
+    config_file: configparser.RawConfigParser,
+    section: str,
+    key: str,
+    default_value: str = "",
+) -> str:
+    if config_file.has_option(section, key):
+        return config_file.get(section, key)
+    return default_value
+
+
+pg_args = {
+    "connect_timeout": "600",
+    "user": "zulip",
+    "dbname": "zulip",
+}
 
-# Path to the root of the Zulip codebase in production
-sys.path.insert(0, "/home/zulip/deployments/current")
-# Path to the root of the Zulip codebase in development
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../..")))
+USING_PGROONGA = False
 try:
+    # Case (1); we insert the path to the development root.
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../..")))
+    # Case (2) could take this branch if it set up the venv first, but
+    # there is no reason to unnecessarily bring in the Django
+    # dependency; it will fall into the path below due the venv not
+    # being configured.
     os.environ["DJANGO_SETTINGS_MODULE"] = "zproject.settings"
     from django.conf import settings
 
-    if settings.REMOTE_POSTGRES_HOST != "":
-        pg_args["host"] = settings.REMOTE_POSTGRES_HOST
-    if settings.REMOTE_POSTGRES_PORT != "":
-        pg_args["port"] = settings.REMOTE_POSTGRES_PORT
+    pg_args["host"] = settings.DATABASES["default"]["HOST"]
+    pg_args["port"] = settings.DATABASES["default"].get("PORT")
+    pg_args["password"] = settings.DATABASES["default"].get("PASSWORD")
+    pg_args["sslmode"] = settings.DATABASES["default"]["OPTIONS"].get("sslmode")
     USING_PGROONGA = settings.USING_PGROONGA
 except ImportError:
-    # process_fts_updates also supports running locally on a remote
-    # PostgreSQL server; in that case, one can just connect to localhost
-    USING_PGROONGA = False
-
-# Since we don't want a hard dependency on being able to access the
-# Zulip settings (as we may not be running on a server that has that
-# data), we determine whether we're using PGroonga using
-# /etc/zulip/zulip.conf.
-#
-# However, we still also check the `USING_PGROONGA` variable, since
-# that's all we have in development.
-config_file = configparser.RawConfigParser()
-config_file.read("/etc/zulip/zulip.conf")
-if config_file.has_option("machine", "pgroonga"):
-    USING_PGROONGA = True
-
-if "host" in pg_args:
-    pg_args["password"] = ""
-    if settings.DATABASES["default"]["PASSWORD"] is not None:
-        pg_args["password"] = settings.DATABASES["default"]["PASSWORD"]
-    pg_args["user"] = settings.DATABASES["default"]["USER"]
-    pg_args["dbname"] = settings.DATABASES["default"]["NAME"]
-    if settings.REMOTE_POSTGRES_SSLMODE != "":
-        pg_args["sslmode"] = settings.REMOTE_POSTGRES_SSLMODE
-    else:
-        pg_args["sslmode"] = "verify-full"
```

Comment:
It seems like you lost this `sslmode` default?

Reply:
This is set on line 316 of `zproject/computed_settings.py` so is no longer needed here.